subsys/espi/ndsx_espi.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/espi.h>
+
+#include "x86_non_dsx.h"
+
+static const struct device *espi_dev;
+static struct espi_callback espi_bus_cb;
+static struct espi_callback espi_chan_cb;
+static struct espi_callback espi_vw_cb;
+
+static void espi_bus_reset(void)
+{
+	/* If SOC is up toggle the PM_PWRBTN pin */
+	if (power_gpio_sig_get(POWER_PCH_EC_SLP_SUS_N)) {
+		LOG_INF("Toggle PM PWRBTN");
+
+		gpio_set_level(EC_PCH_PWRBTN_N, 0);
+		k_msleep(POWER_EC_PCH_PM_PWRBTN_DELAY_MS);
+		power_gpio_sig_set(EC_PCH_PWRBTN_N, 1);
+	}
+}
+
+static void espi_vw_handler(struct espi_event *event)
+{
+	LOG_DBG("VW is triggered, event=%d, val=%d\n", event->evt_details,
+			power_vw_sig_get(event->evt_details));
+
+	switch (event->evt_details) {
+	case ESPI_VWIRE_SIGNAL_SLP_S3:
+	case ESPI_VWIRE_SIGNAL_SLP_S4:
+		k_thread_resume(&pwrseq_thread_id);
+		break;
+	default:
+		break;
+	}
+}
+
+static void espi_bus_handler(const struct device *dev,
```

Comment:
The code in this function seems generic to eSPI. I would expect this to be in the espi driver itself

Reply:
ACK, I will add in follow on CLs